og.module

```diff
@@ -203,17 +224,32 @@ function og_entity_create_access(AccountInterface $account, array $context, $bun
   // @see \Drupal\og\Plugin\EntityReferenceSelection\OgSelection::buildEntityQuery()
   $required = FALSE;
 
+  /** @var \Drupal\og\OgGroupAudienceHelperInterface $audience_helper */
+  $audience_helper = \Drupal::service('og.group_audience_helper');
+  $audience_fields = $audience_helper->getAllGroupAudienceFields($entity_type_id, $bundle);
   $field_definitions = \Drupal::service('entity_field.manager')->getFieldDefinitions($entity_type_id, $bundle);
+
   foreach ($field_definitions as $field_name => $field_definition) {
     /** @var \Drupal\Core\Field\FieldDefinitionInterface $field_definition */
-    if (!\Drupal::service('og.group_audience_helper')->isGroupAudienceField($field_definition)) {
+    if (!in_array($field_definition->getName(), array_keys($audience_fields))) {
       continue;
     }
 
-    $handler = Og::getSelectionHandler($field_definition);
+    if (count($audience_fields) === 1) {
+      // Set the field to required only if we have a single audience field.
+      // If we have multiple fields then the require logic will perform at the
+      // entity constraint level.
+      $field_definition->setRequired(!\Drupal::entityTypeManager()
+        ->getAccessControlHandler($entity_type_id)
+        ->createAccess($bundle, $account, ['skip_og_permission' => TRUE]));
+    }
```

Comment:
Why are we again setting fields to be required inside an access hook? This doesn't belong here.

Reply:
I believe it's related to https://github.com/Gizra/og/pull/570/files#r377001979

Reply:
We should investigate a different solution, this is a hack. We should not alter any field definitions in `hook_entity_access()`.

It looks like this field definition is used as temporary storage for what could be a simple `$is_required` flag.